src/main/java/com/exadel/discount/entity/Discount.java

```diff
@@ -0,0 +1,40 @@
+package com.exadel.discount.entity;
+
+import lombok.*;
+import org.hibernate.annotations.GenericGenerator;
+
+import javax.persistence.*;
+import java.util.Set;
+import java.util.UUID;
+
+@Entity
+@Table(name = "discounts")
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(exclude = {"tags"})
+@ToString(exclude = {"tags"})
+public class Discount {
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(generator = "UUID")
+    @GenericGenerator(
+            name = "UUID",
+            strategy = "org.hibernate.id.UUIDGenerator"
+    )
+    private UUID id;
+
+    @Column(name = "name", nullable = false, length = 50)
+    private String name;
+
+    @Column(name = "promo", nullable = false, length = 50)
+    private String promo;
+
+    @ManyToMany(fetch = FetchType.LAZY)
+    @JoinTable(
+            name = "tags_discounts",
+            joinColumns = @JoinColumn(name = "discount_id"),
+            inverseJoinColumns = @JoinColumn(name = "tag_id")
+    )
+    private Set<Tag> tags;
+}
```

Comment:
add this code in Discount class fo connection with below entities, please:
@OneToOne(mappedBy = "discounts", cascade = CascadeType.REMOVE)
    private Coupon coupon;
@OneToOne(mappedBy = "discounts", cascade = CascadeType.REMOVE)
    private Favorite favorite;

Reply:
no, only after your merge to dev branch, she don't have this entities